File: 22.cpp
/*
    struct ListNode{
        int val;
        ListNode *next;
        ListNode(int x){
            val = x;
            next = nullptr;
        }
    }
*/


ListNode* FindKthToTail(ListNode* ListHead, unsigned int k){
    if(ListHead == nullptr || k <= 0)
        return nullptr;
    ListNode* behind = nullptr, *ahead = ListHead;
    while(k && ahead != nullptr){
        ahead = ahead -> next;
        -- k;
    }
    if(k != 0)
        return nullptr; // 不足k个节点
    behind = ListHead;
    while(ahead != nullptr){
        ahead = ahead -> next;
        behind = behind -> next;
    }
    return behind;
}

File: 22.java
public class Solution {

    public ListNode FindKthToTail(ListNode head, int k) {
        if (head == null || k < 1)
            return null;
        
        ListNode node = head, pre = head;
        while (node != null && k > 0) {
            node = node.next;
            -- k;
        }
        if (k > 0)
            return null;
        while (node != null) {
            pre = pre.next;
            node = node.next;
        }
        return pre;
    }
}
